core/trino-main/src/main/java/io/trino/server/security/oauth2/OAuth2Config.java

```diff
@@ -131,16 +180,17 @@ public OAuth2Config setClientSecret(String clientSecret)
         return this;
     }
 
-    public Optional<String> getAudience()
+    @NotNull
+    public List<String> getAdditionalAudiences()
     {
-        return audience;
+        return additionalAudiences;
     }
 
-    @Config("http-server.authentication.oauth2.audience")
-    @ConfigDescription("The required audience of a token")
-    public OAuth2Config setAudience(String audience)
+    @Config("http-server.authentication.oauth2.additional-audiences")
```

Comment:
Should this be using @LegacyConfig to allow older configs to still work for the time being?

Reply:
This is tricky as the semantics of the two properties aren't quite the same, I could make this change and it would work I believe.

Reply:
I've made this a `@LegacyConfig`